src/Common/MemoryTracker.cpp

```diff
@@ -298,11 +303,3 @@ namespace CurrentMemoryTracker
         }
     }
 }
-
-DB::SimpleActionLock getCurrentMemoryTrackerActionLock()
```

Comment:
This code was wrong, because it disables memory tracking for all threads.
When one thread is throwing exception, another thread was being able to overallocate, presenting the race condition.

I did not find any cases when this type of cancelling is needed.